internal/databroker/server.go

```diff
@@ -452,18 +454,15 @@ func (srv *Server) getBackend() (backend storage.Backend, err error) {
 	return backend, nil
 }
 
-func (srv *Server) newBackendLocked() (backend storage.Backend, err error) {
-	ctx := context.Background()
-
+func (srv *Server) newBackendLocked(ctx context.Context) (storage.Backend, error) {
 	switch srv.cfg.storageType {
 	case config.StorageInMemoryName:
-		log.Ctx(ctx).Info().Msg("using in-memory store")
+		log.Ctx(ctx).Info().Msg("initializing new in-memory store")
 		return inmemory.New(), nil
 	case config.StoragePostgresName:
-		log.Ctx(ctx).Info().Msg("using postgres store")
-		backend = postgres.New(srv.cfg.storageConnectionString)
+		log.Ctx(ctx).Info().Msg("initializing new postgres store")
+		return postgres.New(srv.backendCtx, srv.cfg.storageConnectionString), nil
```

Comment:
Can you help me understand `backendCtx`? Why is this different from the new `ctx` parameter on `getBackend()`?

Reply:
OK thanks, makes sense. Would you mind adding that explanation in a comment in the code?